src/Movepiece.js

```diff
@@ -0,0 +1,55 @@
+let piecePosition = {x: 4, y: 0}
+function MovePiece(direction){ // function that moves the piece
+    switch (direction) {
+        case 'left':
+            piecePosition.x -= 1;
+            break;
+        case 'right':
+            piecePosition.x += 1;
+            break;
+        case 'down':
+            piecePosition.y += 1;
+            break; 
+    }
+}
+
+function collision(){ // checks if pieces are touching and stops them from colliding
+    for(let rows = 0; rows < currentPiece.shape.length; rows++){
+        for(let cols = 0; cols < currentPiece.shape[rows].length; cols++) {
+            if(currentPiece.shape[rows][cols]){
+                const newX = piecePosition.x + cols;
+                const newY = piecePosition.y + rows;
+                if (newX < 0 || newX >= canvas.width/grid || newY >= canvas.height/grids){
```

Comment:
Looking through, I wasn't quite sure where the references to `canvas` & `grid` come from. Closest natural variable seems to be `gridrows` and `gridcols` in index.js